### DISCUSSION TITLE
Adding a lot of markers in cluster takes a while

### DISCUSSION THREAD

**Discussion by FrenchGithubUser** [Leaflet/Leaflet#8347] [Q&A]
I am trying to add a lot (20k+) markers to a leaflet map with the markercluster extension.
I followed the recommendations (used to add 100k markers) but it still takes up to 10s.
I am using leaflet in a vuejs project, idk if it might come from this.

Here is how it is done : 

```
       this.tiles = L.tileLayer(
        "https://tilesprovider.com",
        {
          attribution: "attribution",
        }
      );
      this.map = L.map("map", {
        minZoom: 3,
        zoomControl: false,
        layers: this.tiles,
        center: mapCenter === null ? [47.2, 0] : mapCenter,
        zoom: 10,
        crossOrigin: true,
      });

      this.markers = L.markerClusterGroup();
      this.markerList = [];

      pointsToAdd.forEach((spot) => {
        const marker = L.marker(L.latLng(spot.lat, spot.lon), {
          icon: this.pinIcon,
        }).on("click", () => this.markerClicked(spot, marker));
        this.markerList.push(marker);
        this.spots.push(spot);
      });

    this.markers.addLayers(this.markerList);
    this.map.addLayer(this.markers);
```

Any idea of how to improve the performances ?

**Comment by jonkoops** [comment]
Rendering a lot of markers can lead to performance issues as a DOM element is added for each one of them. I'd recommend you look for a plugin to cluster your markers such as [Leaflet.markercluster](https://github.com/Leaflet/Leaflet.markercluster).

**Reply by FrenchGithubUser** [reply]
yes, I already use this plugin but I am still facing performance issues
